File: grind-75/string/validParenthesis.py
class Solution:
    @classmethod
    def isValid(self, s):
        stack = []
        hmap = {"]": "[", "}": "{", ")": "("}
        for char in s:
            # Add opening brackets to stack
            if char in hmap.values():
                stack.append(char)
            # Check if closing bracket breaks rules
            elif char in hmap.keys():
                if stack == [] or hmap[char] != stack.pop():
                    return False
            else:
                return False
        return stack == []


test_value = "()[]{}"
result = Solution.isValid(test_value)
print(result)

File: projects/grind-75/go/string/valid_parenthesis/valid_parenthesis.go
package util

func isValid(s string) bool {
	stack := []rune{}
	hmap := map[rune]rune{']': '[', '}': '{', ')': '('}

	for _, char := range s {
		if char == '[' || char == '{' || char == '(' {
			stack = append(stack, char)
		} else if char == ']' || char == '}' || char == ')' { // Is closing bracket.
			if len(stack) == 0 || hmap[char] != stack[len(stack)-1] {
				return false
			}
			stack = stack[:len(stack)-1]
		} else {
			return false
		}
	}
	return len(stack) == 0
}
